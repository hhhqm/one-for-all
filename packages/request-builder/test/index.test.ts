import spec from './petstore-spec';
import Builder from '../src/index';

const operations: Record<string, RequestBuilder.RequestParams> = {
  updatePet: {
    body: {}
  },
  addPet: {
    body: {}
  },
  findPetsByStatus: {

  },
  findPetsByTags: {

  },
  getPetById: {
    params: { petId: 'foobar' }
  },
  updatePetWithForm: {
    params: { petId: 'foobar' }
  },
  deletePet: {
    params: { petId: 'foobar' }
  },
  uploadFile: {
    params: { petId: 'foobar' }
  },
  getInventory: {

  },
  placeOrder: {

  },
  getOrderById: {
    params: { orderId: 'foobar' }
  },
  deleteOrder: {
    params: { orderId: 'foobar' }
  },
  createUser: {

  },
  createUsersWithListInput: {

  },
  loginUser: {

  },
  logoutUser: {

  },
  getUserByName: {
    params: { username: 'foobar' }
  },
  updateUser: {
    params: { username: 'foobar' }
  },
  deleteUser: {
    params: { username: 'foobar' }
  },
}

test('expect to match snapshots', () => {
  const builder = new Builder(spec as any);
  Object.entries(operations).forEach(([operationId, params]) => {
    expect(builder.fillRequest(operationId, params)).toMatchSnapshot();
  })
})
