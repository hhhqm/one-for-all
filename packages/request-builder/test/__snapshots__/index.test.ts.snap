// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to match snapshots 1`] = `
Object {
  "body": Object {},
  "method": "put",
  "path": "/pet",
}
`;

exports[`expect to match snapshots 2`] = `
Object {
  "body": Object {},
  "method": "post",
  "path": "/pet",
}
`;

exports[`expect to match snapshots 3`] = `
Object {
  "method": "get",
  "path": "/pet/findByStatus",
  "query": Object {
    "\${p": Object {
      "name}": undefined,
    },
  },
}
`;

exports[`expect to match snapshots 4`] = `
Object {
  "method": "get",
  "path": "/pet/findByTags",
  "query": Object {
    "\${p": Object {
      "name}": undefined,
    },
  },
}
`;

exports[`expect to match snapshots 5`] = `
Object {
  "method": "get",
  "path": "/pet/{petId}",
}
`;

exports[`expect to match snapshots 6`] = `
Object {
  "method": "post",
  "path": "/pet/{petId}",
  "query": Object {
    "\${p": Object {
      "name}": undefined,
    },
  },
}
`;

exports[`expect to match snapshots 7`] = `
Object {
  "header": Object {
    "\${p": Object {
      "name}": undefined,
    },
  },
  "method": "delete",
  "path": "/pet/{petId}",
}
`;

exports[`expect to match snapshots 8`] = `
Object {
  "body": undefined,
  "method": "post",
  "path": "/pet/{petId}/uploadImage",
  "query": Object {
    "\${p": Object {
      "name}": undefined,
    },
  },
}
`;

exports[`expect to match snapshots 9`] = `
Object {
  "method": "get",
  "path": "/store/inventory",
}
`;

exports[`expect to match snapshots 10`] = `
Object {
  "body": undefined,
  "method": "post",
  "path": "/store/order",
}
`;

exports[`expect to match snapshots 11`] = `
Object {
  "method": "get",
  "path": "/store/order/{orderId}",
}
`;

exports[`expect to match snapshots 12`] = `
Object {
  "method": "delete",
  "path": "/store/order/{orderId}",
}
`;

exports[`expect to match snapshots 13`] = `
Object {
  "body": undefined,
  "method": "post",
  "path": "/user",
}
`;

exports[`expect to match snapshots 14`] = `
Object {
  "body": undefined,
  "method": "post",
  "path": "/user/createWithList",
}
`;

exports[`expect to match snapshots 15`] = `
Object {
  "method": "get",
  "path": "/user/login",
  "query": Object {
    "\${p": Object {
      "name}": undefined,
    },
  },
}
`;

exports[`expect to match snapshots 16`] = `
Object {
  "method": "get",
  "path": "/user/logout",
}
`;

exports[`expect to match snapshots 17`] = `
Object {
  "method": "get",
  "path": "/user/{username}",
}
`;

exports[`expect to match snapshots 18`] = `
Object {
  "body": undefined,
  "method": "put",
  "path": "/user/{username}",
}
`;

exports[`expect to match snapshots 19`] = `
Object {
  "method": "delete",
  "path": "/user/{username}",
}
`;
